### ISSUE TITLE
trial tracebacks (--tbformat emacs) don't supress internal method calls properly

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7576]
|[<img alt="warner's avatar" src="https://avatars.githubusercontent.com/u/27146?s=50" width="50" height="50">](https://github.com/warner)| @warner reported|
|-|-|
|Trac ID|trac#339|
|Type|enhancement|
|Created|2003-10-16 00:30:08Z|

Attachments:

* [filefile](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/6c4/6c41101fe24a8f80c8cb51781f9e9ecf8c7a4d39/c90ec4ab6fea6e24ab4bb458503403a55c79831c) (1800 bytes) - added by moshez on 2003-10-19 13:00:03Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__339 339
type__enhancement enhancement
reporter__warner warner
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1066264208000000 1066264208000000
changetime__1469004504171932 1469004504171932
version__ 
owner__ 
cc__glyph cc__warner cc__jml
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="warner's avatar" src="https://avatars.githubusercontent.com/u/27146?s=50" width="50" height="50">](https://github.com/warner)|@warner commented|
|-|-|

```
#!html
<pre>
Once upon a time, trial would remove its internal method
calls from the traceback it displays any time an ERROR or
FAILURE occurred in the test case. I remember fixing this
code at one point, many moons ago, and I know there is a
unit test present for this feature.

I've noticed that this frame supression isn't working any
more. The unit test still passes, but actual trial output is
noisier than it ought to be.

A sample output is below: it ought to show just a single
frame, the test_threads.py one. Instead it shows three above
and one below the actual error. (it is likely that the frame
supression code was only meant to remove the three above, but
I think the output would be easier to read if it removed the
unittest.py:fail() frame as well)

FAILURE: testCallInThread (twisted.test.test_threads.ThreadsTestCase)
-------------------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/slave/twisted/bsd-full2.2/Twisted/twisted/trial/unittest.py", line
249, in _runPhase
    stage(*args, **kwargs)
  File "/home/slave/twisted/bsd-full2.2/Twisted/twisted/trial/unittest.py", line
271, in _main
    self.runner(self.method)
  File "/home/slave/twisted/bsd-full2.2/Twisted/twisted/trial/runner.py", line
102, in runTest
    method()
  File "/home/slave/twisted/bsd-full2.2/Twisted/twisted/test/test_threads.py",
line 90, in testCallInThread
    self.fail("threads lost a count")
  File "/home/slave/twisted/bsd-full2.2/Twisted/twisted/trial/unittest.py", line
98, in fail
    raise FailTest, message
FailTest: threads lost a count
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
Here's a patch which fixes it, at least for the non-emacs case.
The problem was that the unit test was testing a function that worked
perfectly -- but was not called. So I just had to call it :)
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
Marking as "PATCH" and assigning to warner.
Please test and apply.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="warner's avatar" src="https://avatars.githubusercontent.com/u/27146?s=50" width="50" height="50">](https://github.com/warner)|@warner commented|
|-|-|

```
#!html
<pre>
patch tested and committed (with tweaks).
It'd be nice if the emacs formatter could do the same kind of
"hide internals" magic, but it wasn't immediately obvious to me
how to accomplish that. I'm marking this as "done- could be better"
to indicate that util.emacs_format_traceback should be improved.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
/cvs/Twisted/twisted/trial/util.py,v  &lt;--  util.py
new revision: 1.7; previous revision: 1.6
Fixed
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph commented|
|-|-|

```
#!html
<pre>
Magical internal mangling considered harmful.  I am reverting this
information-loss because it is killing my productivity while using Trial.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
Please either kill this bug off or explain what you want to be done.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
I'm downgrading this to feature, because it is
obviously helpful for some people to have the current
behaviour
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph commented|
|-|-|

```
#!html
<pre>
This is never going to happen.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_11"></a>|GitHub commented|
|-|-|

In changeset 2049661a6530fa03c34dead92c3dbbb5cacb5c89
```
#!CommitTicketReference repository="" revision="2049661a6530fa03c34dead92c3dbbb5cacb5c89"
Merge pull request #339 from twisted/8607-rodrigc-zipfile

Author: rodrigc
Reviewer: acabhishek942
Fixes: #8607

Port zipstream to Python 3
```
